hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestCloseContainer.java

```diff
@@ -159,11 +183,72 @@ public void testCloseClosedContainer()
     StorageContainerManager scm = cluster.getStorageContainerManager();
     // Pick any container on the cluster and close it via client
     ContainerInfo container = scm.getContainerManager().getContainers().get(0);
+    // Checksum file doesn't exist before container close
+    List<HddsDatanodeService> hddsDatanodes = cluster.getHddsDatanodes();
+    for (HddsDatanodeService hddsDatanode: hddsDatanodes) {
+      assertFalse(containerChecksumFileExists(hddsDatanode, container));
+    }
+    // Close container
     OzoneTestUtils.closeContainer(scm, container);
+
+    // Checksum file exists after container close
+    for (HddsDatanodeService hddsDatanode: hddsDatanodes) {
+      GenericTestUtils.waitFor(() -> waitForContainerChecksumToExist(hddsDatanode, container), 100, 5000);
+      assertTrue(containerChecksumFileExists(hddsDatanode, container));
+    }
+
     assertThrows(IOException.class,
         () -> cluster.getStorageContainerLocationClient()
             .closeContainer(container.getContainerID()),
         "Container " + container.getContainerID() + " already closed");
   }
 
+  @Test
+  public void testContainerChecksumForClosedContainer() throws Exception {
+    // Create some keys to write data into the open containers
+    for (int i = 0; i < 10; i++) {
+      TestDataUtil.createKey(bucket, "key" + i, ReplicationFactor.THREE,
+          ReplicationType.RATIS, "this is the content");
+    }
+    StorageContainerManager scm = cluster.getStorageContainerManager();
+    ContainerInfo containerInfo = scm.getContainerManager().getContainers().get(0);
+    // Checksum file doesn't exist before container close
+    List<HddsDatanodeService> hddsDatanodes = cluster.getHddsDatanodes();
+    for (HddsDatanodeService hddsDatanode : hddsDatanodes) {
+      assertFalse(containerChecksumFileExists(hddsDatanode, containerInfo));
+    }
+
+    // Build expected merkle tree
+    HddsDatanodeService hddsDatanodeService = cluster.getHddsDatanodes().get(0);
+    OzoneContainer ozoneContainer = hddsDatanodeService.getDatanodeStateMachine().getContainer();
+    KeyValueContainer container = (KeyValueContainer) ozoneContainer.getController()
+        .getContainer(containerInfo.getContainerID());
+    ContainerProtos.ContainerMerkleTree expectedMerkleTree = buildContainerMerkleTree(
+        container.getContainerData(), cluster.getConf());
```

Comment:
Since the test method is basically a copy/paste of the actual method, I'm not sure it's doing much for us here. `TestContainerMerkleTree` already has tests on the exact tree constructed. At the integration test level it would probably be better to test the trees relative to each other and the container itself:
- Test trees of different replicas on the same container are equal.
- Test trees of different containers are different.
- Test block and chunk IDs and counts in trees match those in the containers.